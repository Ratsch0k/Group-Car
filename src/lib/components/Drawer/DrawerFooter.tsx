import React from 'react';
import {Grid, Link} from '@material-ui/core';
import useModalRouter from 'lib/hooks/useModalRouter';
import {useTranslation} from 'react-i18next';

type GridProps = import('@material-ui/core').GridProps;

interface DrawerFooterProps extends GridProps {
  justify?: undefined;
}

const DrawerFooter: React.FC<DrawerFooterProps> =
(props: DrawerFooterProps) => {
  const {...rest} = props;
  const {goTo} = useModalRouter();
  const {t} = useTranslation();

  return (
    <Grid container {...rest} justify='space-evenly'>
      <Grid item>
        <Link component='button' onClick={() => goTo('/privacy-policy')}>
          {t('privacyPolicy.title')}
        </Link>
      </Grid>
      <Grid item>
        <Link component='button' onClick={() => goTo('/imprint')}>
          {t('imprint.title')}
        </Link>
      </Grid>
    </Grid>
  );
};

export default DrawerFooter;
