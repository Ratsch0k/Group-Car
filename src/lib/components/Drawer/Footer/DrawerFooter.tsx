import React from 'react';
import {Grid, Link, Container} from '@material-ui/core';
import {
  useModalRouter,
  DrawerCard,
  GroupCarTheme,
} from 'lib';
import {useTranslation} from 'react-i18next';
import {makeStyles, createStyles} from '@material-ui/styles';
import clsx from 'clsx';


type GridProps = import('@material-ui/core').GridProps;

/**
 * Props.
 */
interface DrawerFooterProps extends GridProps {
  justify?: undefined;
  className?: string;
}


/**
 * Element.
 */
export const DrawerFooter: React.FC<DrawerFooterProps> =
(props: DrawerFooterProps) => {
  const {goTo} = useModalRouter();
  const {t} = useTranslation();

  /**
 * Styles
 */
  const useStyles = makeStyles((theme: GroupCarTheme) =>
    createStyles({
      paper: {
        width: '100%',
      },
      container: {
        padding: theme.spacing(1),
      },
    }),
  );
  const classes = useStyles();

  return (
    <DrawerCard
      className={clsx(props.className, classes.paper)}
    >
      <Container className={classes.container}>
        <Grid container justify='space-evenly'>
          <Grid item>
            <Link component='button' onClick={() => goTo('/privacy-policy')}>
              {t('privacyPolicy.title')}
            </Link>
          </Grid>
          <Grid item>
            <Link component='button' onClick={() => goTo('/imprint')}>
              {t('imprint.title')}
            </Link>
          </Grid>
        </Grid>
      </Container>
    </DrawerCard>
  );
};

export default DrawerFooter;
