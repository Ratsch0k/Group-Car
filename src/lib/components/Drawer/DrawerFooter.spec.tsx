import React from 'react';
import {render} from '@testing-library/react';
import DrawerFooter from './DrawerFooter';

it('renders without crashing', () => {
  render(<DrawerFooter />);
});

it('matches snapshot', () => {
  const {container} = render(<DrawerFooter />);

  expect(container).toMatchSnapshot();
});
