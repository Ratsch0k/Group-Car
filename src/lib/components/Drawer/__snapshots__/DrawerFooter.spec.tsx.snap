// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <button
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
      >
        Datenschutzerkl√§rung
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <button
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
      >
        Impressum
      </button>
    </div>
  </div>
</div>
`;
